source/commands/danger-pr.ts

```diff
@@ -1,5 +1,50 @@
 import * as program from "commander"
+import * as debug from "debug"
+import * as fs from "fs"
+import { FakeCI } from "../ci_source/providers/Fake"
+import { GitHub } from "../platforms/GitHub"
+import { Executor } from "../runner/Executor"
+import { pullRequestParser } from "../platforms/github/pullRequestParser"
+import { runDangerfileEnvironment } from "../runner/DangerfileRunner"
 
-program.parse(process.argv)
+const d = debug("danger:pr")
 
-console.log("Not Yet implmented")
+program
+  .option("-d, --dangerfile [filePath]", "Specify custom dangefile other than default dangerfile.js")
+  .parse(process.argv)
+
+const dangerFile = (program as any).dangerfile || "dangerfile.js"
+const stat = fs.statSync(dangerFile)
+
+if (program.args.length === 0) {
+  console.error("Please include a PR URL to run against")
+  process.exitCode = 1
+} else {
+  const pr = pullRequestParser(program.args[0])
+
+  if (!pr) {
+    console.error("Could not get a repo and a PR number from your URL, bad copy & paste?")
+    process.exitCode = 1
+  } else {
+
+    const source = new FakeCI({
+      DANGER_TEST_REPO: pr.repo,
+      DANGER_TEST_PR: pr.pullRequestNumber
+    })
+
+    const platform = new GitHub(null, source)
```

Comment:
It would be cool to read in a GitHub API token somehow for testing against private repo PR URLs. 

Perhaps the user should pass `DANGER_GITHUB_API_TOKEN` as an environment variable when executing the `danger pr` command, or perhaps `danger pr` could support a `--token` option. What do you think?